### DISCUSSION TITLE
math expressions ignored in incremental slides

### DISCUSSION THREAD

**Discussion by JosiahParry** [quarto-dev/quarto-cli#3401] [Q&A] [answered]
When including a math expression with `$$ ... $$` they appear immediately and are ignored by incremental options.

tag @cscheid 

reprex below

```quarto
---
title: "incremental formulae"
format:
  revealjs:
    incremental: true   
---

# I'm probably doing this wrong

## A slide

- formula 1 (lol)

$$
G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
{\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
$$

- formula 2

$$
Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
$$


## Another one 


:::{.incremental}
- formula 1 (lol)

$$
G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
{\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
$$

- formula 2

$$
Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
$$

:::
```

**Comment by cscheid** [comment, marked as answer]
The trouble you're having is that these are incremental _lists_, and not incremental _slides_ (it's confusing, and it confused me too! But the feature comes ultimately from pandoc and we don't truly control it).

If you want to achieve the effect you're looking for, use "pauses" (`. . .`):

````
---
title: "incremental formulae"
format: revealjs
---

## A slide

. . .

- formula 1 (lol)

$$
G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
{\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
$$

. . .

- formula 2

$$
Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
$$
````

**Comment by mcanouil** [comment]
The issue is that your math equations are outside of the items in the list.
You need to properly indent your math.

PS: could you change the category from "general" to "Q&A" and when satisfied with a reply mark it as the answer to let others know this has been answered? This can only be achieved using a web browser in "desktop mode".

https://user-images.githubusercontent.com/8896044/202922406-1c4e744b-b03e-452d-8041-3841301c78a8.mov

````
---
title: "incremental formulae"
format:
  revealjs:
    incremental: true   
---

# I'm probably doing this wrong

## A slide

- formula 1 (lol)
  $$
  G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
  {\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
  $$

- formula 2
  $$
  Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
  $$

## Another one 

:::{.incremental}
- formula 1 (lol)
  $$
  G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
  {\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
  $$

- formula 2
  $$
  Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
  $$
:::
````

**Reply by JosiahParry** [reply]
As @cscheid pointed out, it's a limitation of pandoc. The name `incremental` is misleading as its not entirely incremental but only incremental for lists so maybe they should've called it `incremental_lists`.

The centering of the equation was desired. 

I guess "pausing" is the phrase that I'm after.

**Reply by cscheid** [reply]
@mcanouil That's different markup from what @JosiahParry had, though. What you've done turns the syntax into a _single list_, and then `incremental` works. See this: https://github.com/rstudio/revealjs/issues/70

**Reply by mcanouil** [reply]
@JosiahParry Meaning you want your slide to appear in four steps, not in two steps?
(edit: @cscheid the above part was not clear, hence my answer with a slightly different structure/syntax)

Because if the issue is only the centering, then the following will fix that (and maybe it should be the default? @cscheid)

````
---
title: "incremental formulae"
format:
  revealjs:
    incremental: true
    include-in-header: 
      - text: |
          <style>.reveal ol, .reveal dl, .reveal ul { display: block; }</style>
---

# I'm probably doing this wrong

## A slide

- formula 1 (lol)

  $$
  G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
  {\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
  $$

- formula 2

  $$
  Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
  $$


## Another one 


:::{.incremental}
- formula 1 (lol)

  $$
  G = \frac{\sum_{i = 1}^{n} \sum_{j = 1}^{n} w_{ij}x_ix_j}
  {\sum_{i = 1}^{n}\sum_{j = 1}^{n} x_i x_j}
  $$

- formula 2

  $$
  Gi = \frac{\sum_j w_{ij} x_j}{\sum_j x_j}
  $$

:::

````


https://user-images.githubusercontent.com/8896044/202923954-b50623dd-9878-47e8-a87f-ac17182a03fe.mov

**Reply by JosiahParry** [reply]
Pauses are still the answer from my POV. I avoid CSS at all costs! 🤣
